### PR TITLE
MAT-79, MAT-81

### PR DESCRIPTION
**Pull request by scarabus**

### CODE REVIEW

File: Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java [hunk 1]
```diff
@@ -42,26 +42,47 @@
...
+        this((Long) null, DateTimeZone.UTC);
+    }
+
+    public ReflexTimeValue(Long initial, DateTimeZone timezone) {
+        this.timezone = timezone;
+        this.dateTime = (initial == null) ? new DateTime(timezone) : new DateTime(initial, timezone);
+        timeFormat = DateTimeFormat.forPattern("HH:mm:ss").withZone(timezone);
+
+        long l = dateTime.getMillis();
+        System.out.println("MILLIS = " + l);
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java]: use log

File: Libs/Reflex/src/main/java/reflex/value/ReflexDateValue.java [hunk 2]
```diff
@@ -70,7 +70,7 @@ public ReflexDateValue(LocalDate other, String calendar) {
         setupCalendarHandler();
     }
 
-    private static final DateTimeFormatter FORMATTER = DateTimeFormat.forPattern("yyyyMMdd").withZoneUTC();
+    private DateTimeFormatter FORMATTER = DateTimeFormat.forPattern("yyyyMMdd").withZoneUTC();
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/value/ReflexDateValue.java]: why the removal of static final?

File: Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java [hunk 3]
```diff
@@ -42,26 +42,47 @@
 
     private DateTime dateTime;
     private DateTimeFormatter timeFormat = DateTimeFormat.forPattern("HH:mm:ss").withZoneUTC();
+    DateTimeZone timezone = DateTimeZone.UTC;
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java]: package scope?

File: Libs/Reflex/src/main/java/reflex/function/TimeNode.java [hunk 4]
```diff
@@ -41,29 +44,40 @@
  */
 public class TimeNode extends BaseNode {
     private ReflexNode initExpression;
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/function/TimeNode.java]: is this variable still used?

File: Libs/Reflex/src/main/java/reflex/function/TimeNode.java [hunk 5]
```diff
@@ -41,29 +44,40 @@
...
-    public TimeNode(int lineNumber, IReflexHandler handler, Scope s, ReflexNode expression) {
-        super(lineNumber, handler, s);
-        this.initExpression = expression;
+    public TimeNode(int lineNumber, IReflexHandler handler, Scope scope, List<ReflexNode> eList) {
+        super(lineNumber, handler, scope);
+        if (eList == null) {
+            this.exprList = new ArrayList<>();
+        } else {
+            this.exprList = eList;
+        }
```
Comment (dukenguyen) [inline Libs/Reflex/src/main/java/reflex/function/TimeNode.java]: thse 5 lines can be shortened to 3, if you just initialize the member variable to an empty array list and then only set it if it is not null

File: Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java (hunk 1 above)
Comment (scarabus) [inline Libs/Reflex/src/main/java/reflex/value/ReflexTimeValue.java]: Removed; shouldn't have been there

File: Libs/Reflex/src/main/java/reflex/value/ReflexDateValue.java (hunk 2 above)
Comment (scarabus) [inline Libs/Reflex/src/main/java/reflex/value/ReflexDateValue.java]: I'm concerned about the use of UTC in this class. It means that the date changes in the middle of the afternoon. We should be able to specify the timezone, which means the formatter shouldn't be a static final. I'll revert it and log a ticket.

**Pull request merged**